packages/backend-defaults/migrations/scheduler/20240712211735_nullable_next_run.js

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// @ts-check
+
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = async function up(knex) {
+  await knex.schema.alterTable('backstage_backend_tasks__tasks', table => {
+    table.setNullable('next_run_start_at');
+  });
+};
+
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.down = async function down(knex) {
+  await knex.schema.alterTable('backstage_backend_tasks__tasks', table => {
+    table.dropNullable('next_run_start_at');
```

Comment:
nit: what happens if there are rows with this column set to null? Can we do anything safe here to get back to a state where this column can't be null?

Reply:
@timbonicus this reminds me - please move over https://github.com/backstage/backstage/blob/master/packages/backend-tasks/src/migrations.test.ts to backend-defaults and add a test that migrates up, adds some data, and then migrates down again

Reply:
👍 I copied over the `migrations.test.ts` and added a test. To Mike's point, I added a delete to the knex `down` to remove any records with a null `next_run_start_at`, reasoning that these must logically have been added after the initial migration.

Reply:
It's also safe to do, as immediately after, the service starts up and starts persisting these descriptors again with whatever is that code's blessed format